modules/core/src/main/java/org/apache/ignite/internal/processors/cache/binary/CacheObjectBinaryProcessorImpl.java

```diff
@@ -533,6 +536,11 @@ public GridBinaryMarshaller marshaller() {
         binaryCtx.updateMetadata(typeId, meta, false);
     }
 
+    /** {@inheritDoc} */
+    @Override public BinaryTypeWriter createBinaryWriter(String igniteWorkDir) {
+        return new BinaryMetadataFileStore(metadataLocCache, ctx, log, igniteWorkDir, binaryMetadataFileStoreDir);
```

Comment:
There will be problems if `binaryMetadataFileStoreDir` is not null in `CacheObjectBinaryProcessorImpl`

Reply:
Also, you don't need `metadataLocCache` for snapshot creation, I think it's better to create `BinaryMetadataFileStore` directly using constructor in snapshot manager and remove this method.

Reply:
Discussed privately and fixed.